### DISCUSSION TITLE
How to mutate data before fill form in filament 3

### DISCUSSION THREAD

**Discussion by kinjal1298** [filamentphp/filament#13215] [Help] [answered]
### Package

Form builder

### Package Version

v3.2

### How can we help you?

I'm attempting to modify the category name before filling out a form in my admin panel. Specifically, I have two tables, "Post" and "Category," connected with a one-to-one relationship. Within the form, I utilize a TextInput field for the category. However, I've encountered an issue: when attempting to utilize the mutateFormDataBeforeFill() method on the edit page, it only retrieves the details of the post, omitting the category information.

Please suggest any potential solutions

**Comment by leandrocfe** [comment, marked as answer]
What about afterStateHydrated?
https://filamentphp.com/docs/3.x/forms/advanced#field-hydration

**Reply by kinjal1298** [reply]
Thank you, @leandrocfe. It worked perfectly for me. I appreciate your help!
